go/ephemeral/lib.go

```diff
@@ -232,12 +232,16 @@ func (e *EKLib) NewUserEKNeeded(ctx context.Context) (needed bool, err error) {
 func (e *EKLib) newUserEKNeeded(ctx context.Context, merkleRoot libkb.MerkleRoot) (needed bool, err error) {
 	defer e.G().CTraceTimed(ctx, fmt.Sprintf("newUserEKNeeded: %v", needed), func() error { return err })()
 
-	// Let's see what the latest server statement is.
+	// Let's see what the latest server statement is. This verifies that the
+	// latest statement was signed by the latest PUK and otherwise fails with
+	// wrongKID set.
```

Comment:
Let's make sure the caller is super clear about relying on this guarantee.

Reply:
done